### PR TITLE
solved issue-#4783

### PR DESCRIPTION
**Pull request by mbisurgi-designfreed**
Hi, I debug and solved the issue #4783. Now it is possible to do 
`var deLocaleData = moment();` 
`deLocaleData.locale('de');`
`deLocaleData.localeData().weekdays(true) // --> ['Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag', 'Samstag', 'Sonntag']`
`deLocaleData.localeData().weekdays() // --> ['Sonntag', 'Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag', 'Samstag']`

Hopes this helps out! Bye

### CODE REVIEW

**Comment by mbisurgi-designfreed** [comment]
Travis failed on locale/ka.js - which was untouched by this PR. I notice some others PR had the same problem. Also, there seems to be a problem with job 4282.6 related to installing dependencies. Thanks

File: src/test/moment/locale.js [hunk 1]
```diff
@@ -504,6 +504,18 @@ test('when in strict mode with inexact parsing, treat periods in min-weekdays li
     assert.equal(moment('thurs', 'dd', true).format('dddd'), 'Thursday');
 });
 
+test('when weekdays is passed \'true\' as argument, days have to be ordered based on locale first day of the week', function (assert) {
+    var deLocaleData = moment();
```
Comment (sdevalapurkar) [inline src/test/moment/locale.js]: nit/question: any reason we can't use `const` here?

File: src/test/moment/locale.js [hunk 2]
```diff
@@ -504,6 +504,18 @@ test('when in strict mode with inexact parsing, treat periods in min-weekdays li
...
+    deLocaleData.locale('de');
+
+    var neLocaleData = moment();
```
Comment (sdevalapurkar) [inline src/test/moment/locale.js]: same here, can we use `const` instead of `var`?

File: src/test/moment/locale.js (hunk 1 above)
Comment (marwahaha) [reply src/test/moment/locale.js]: I think the rest of this library is written in ES5, but `const` was a ES2015 addition:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const

**Comment by marwahaha** [comment]
Thanks for opening this request.
For now, we will not take this change.

I instead merged https://github.com/moment/moment/pull/4553 .

**Pull request closed without merging**